File: 283MoveZeroes/submit.py
class Solution(object):
# 思路1
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        nonZeros = []
        for n in nums:
            if n != 0:
                nonZeros.append(n)
        
        nums[:len(nonZeros)] = nonZeros
        
        nums[len(nonZeros):] = [0] * (len(nums) - len(nonZeros))
            
        

File: 283MoveZeroes/submit.java
public class Solution {
    // 思路2
    public void moveZeroes(int[] nums) {
        // space O(1)
        // time O(n)
        int k = 0; //在[0, ... k) 区间内存储所有非0元素
        for(int i = 0; i < nums.length; i++){
            if (nums[i] != 0) {
                nums[k] = nums[i];
                k++;
            }
        }
        for(int j = k;j < nums.length; j++){
            nums[j] = 0;
        }
    }
}